### ISSUE TITLE
Delayqueue disorders the redis stream data during master restart

### ISSUE THREAD

**Issue by simhan95** [luin/ioredis#1372]
In my setup, where one application (producer) adds data into redis stream using XADD and another application (consumer) data through XREAD and latest ioredis client was used.

During a redis master restart case and producer side, we see below things:
ioredis throws error as "MOVED" and "CLUSTER IS DOWN". On such errors, it queues the events received in delayqueue in one order. On multiple retries during sendcommand, the error is throws and delayQueue is rearranged with events.

When redis connection is back within 3 secs, on a bulk the delayqueue events are sent to redis but in disorder fashion.

On consumer side: we notice events not received in the order.

After trying out couple of options, we still not yet solved this case.
What options were tried out so far:
1. enableOfflineQueue - set this to false
2. maxRedirections - set this to -1 (but those events were rejected during redis restart which is not expected here)
3. retryDelayOnClusterDown - we can't set this option because in production we don't know for how long cluster is down

Did we miss out anything else? 

TL;DR:
 Does ioredis handles delayqueue in effective way such that the events pushed into the queue are sent out of the queue in the same order?

**Comment by simhan95** [comment]
Do we have any updates on this issue?

**Comment by simhan95** [comment]
@luin, can you please help me solve this case? Have anyone faced this before?
